### ISSUE TITLE
missing actionDispatcherBefore in module installation hooks

### ISSUE THREAD

**Issue by kosmaa** [Oksydan/is_themecore#20]
Is:
``` php       
public const HOOKS = [
        'actionOutputHTMLBefore',
        'actionDispatcherAfter',
        'actionFrontControllerSetMedia',
        'displayListingStructuredData',
        'displayHeader',
        'actionProductSearchAfter',
    ];
```
Should be:
```php    
public const HOOKS = [
        'actionOutputHTMLBefore',
        'actionDispatcherAfter',
        'actionDispatcherBefore', //missing
        'actionFrontControllerSetMedia',
        'displayListingStructuredData',
        'displayHeader',
        'actionProductSearchAfter',
    ];`

without it whole starter theme is broken because smarty plugins are not defined

---
a przy okazji dzięki za świetny szablon startowy i moduły. 
pozdrawiam

**Comment by Kandax86** [comment]
Hi,
I think the issue is also a mistake in the upgrade script to pass version

there are 2 "return" in the function 

```
return $module->registerHook('actionDispatcherBefore');
return $module->registerHook('actionOutputHTMLBefore');
```

instead of 

```
return $module->registerHook('actionDispatcherBefore') && 
	$module->registerHook('actionOutputHTMLBefore');
```

**Comment by Oksydan** [comment]
Hi @kosmaa ,

you are right I replaced `actionDispatcherBefore` to `actionDispatcherAfter` and forget about it. Some OPC modules are making redirect and die() inside `actionDispatcher` module. 
Due to that smarty functions were never registered in that case.

I will fix it in version 2.2.1. Thank you for your report.

**Comment by Oksydan** [comment]
Hi @kosmaa and @Kandax86,

new version 2.2.1 has been released.
